radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/p2p/PendingOutboundChannelsManager.java

```diff
@@ -133,18 +137,16 @@ private void handlePeerConnected(PeerEvent.PeerConnected peerConnected) {
     }
   }
 
-  private void handlePeerHandshakeFailed(PeerEvent.PeerHandshakeFailed peerHandshakeFailed) {
+  private void handlePeerHandshakeFailed(PeerHandshakeFailed peerHandshakeFailed) {
     synchronized (lock) {
       peerHandshakeFailed
           .channel()
           .getUri()
+          .map(RadixNodeUri::getNodeId)
+          .flatMap(nodeId -> Optional.ofNullable(this.pendingChannels.remove(nodeId)))
           .ifPresent(
-              uri -> {
-                final var maybeFuture = this.pendingChannels.remove(uri.getNodeId());
-                if (maybeFuture != null) {
-                  maybeFuture.completeExceptionally(new IOException("Peer connection failed"));
-                }
-              });
+              maybeFuture ->
```

Comment:
it's no longer `maybe`, the future is always there :)

Reply:
Fixed